debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/connection/PostgresReplicationConnection.java

```diff
@@ -230,6 +233,30 @@ private void processWarnings(final boolean forced) throws SQLException {
         };
     }
 
+    private PGReplicationStream startPgReplicationStream(final LogSequenceNumber lsn, Function<ChainedLogicalStreamBuilder, ChainedLogicalStreamBuilder> configurator) throws SQLException {
+        assert lsn != null;
+        ChainedLogicalStreamBuilder streamBuilder = pgConnection()
+                .getReplicationAPI()
+                .replicationStream()
+                .logical()
+                .withSlotName(slotName)
+                .withStartPosition(lsn);
+        streamBuilder = configurator.apply(streamBuilder);
+
+        if (statusUpdateIntervalMillis != null && statusUpdateIntervalMillis > 0) {
+            streamBuilder.withStatusInterval(statusUpdateIntervalMillis, TimeUnit.MILLISECONDS);
+        }
+
+        PGReplicationStream stream = streamBuilder.start();
+        // Needed by tests when connections are opened and closed in a fast sequence
+        try {
```

Comment:
Uff, can we move that to test code if it's really needed?

Reply:
I'd love to but this turned out to be the only reliable solution. Again as RDS moves to the plugin that provides the metadata and we will not need to execute double start in fast susccession (need for RDS compatibility test now) the code will be eleiminated.